lib/src/components/NxTextInput/stateful/__tests__/NxStatefulTextInput.test.tsx

```diff
@@ -4,148 +4,288 @@
-    const component = getShallowComponent({ defaultValue: 'foo' });
+  it('sets ref on the input', function() {
+    const ref = React.createRef<HTMLDivElement>(),
+        el = renderEl({ ref });
 
-    expect(component).toHaveProp('value', 'foo');
-    expect(component).toHaveProp('isPristine', true);
+    expect(ref.current).toBe(el);
   });
 
-  it('sets isPristine to false when the onChange handler fires', function() {
-    const component = getShallowComponent({ defaultValue: 'foo' });
-
-    component.prop('onChange')('bar');
+  it('adds specified classNames to the top-level element in addition to the defaults', function() {
+    const el = renderEl({ className: 'foo' }),
+        defaultEl = renderEl()!;
 
-    component.update();
+    expect(el).toHaveClass('foo');
 
-    expect(component).toHaveProp('isPristine', false);
+    for (const cls of Array.from(defaultEl.classList)) {
+      expect(el).toHaveClass(cls);
+    }
   });
 
-  it('updates the values when the onChange handler fires', function() {
-    const component = getShallowComponent({ defaultValue: 'foo' });
+  it('passes additional attrs to the input', function() {
+    const input = quickRender({ id: 'foo', lang: 'en-US' }).getByRole('textbox');
 
-    component.prop('onChange')('bar');
-
-    component.update();
+    expect(input).toHaveAttribute('id', 'foo');
+    expect(input).toHaveAttribute('lang', 'en-US');
+  });
 
-    expect(component).toHaveProp('value', 'bar');
+  it('sets the value as specified', function() {
+    expect(quickRender({ defaultValue: 'boo' }).getByRole('textbox')).toHaveAttribute('value', 'boo');
   });
 
-  it('passes onChange events to its own onChange prop', function() {
-    const onChange = jest.fn(),
-        component = getShallowComponent({ defaultValue: 'foo', onChange });
+  it('calls onChange with the value whenever the input\'s onChange event fires', async function() {
+    const user = userEvent.setup(),
+        onChange = jest.fn(),
+        input = quickRender({ onChange }).getByRole('textbox');
 
     expect(onChange).not.toHaveBeenCalled();
 
-    component.prop('onChange')('bar');
+    await user.type(input, 'a');
 
-    expect(onChange).toHaveBeenCalledWith('bar');
+    expect(onChange).toHaveBeenCalledWith('a');
   });
 
-  it('passes onKeyPress events to its own onKeyPress prop', function() {
-    const onKeyPress = jest.fn(),
-        component = getShallowComponent({ defaultValue: 'foo', onKeyPress });
+  it('calls onKeyPress with the key value whenever the input\'s onKeyPress event fires', async function() {
+    const user = userEvent.setup(),
+        onKeyPress = jest.fn(),
+        input = quickRender({ onKeyPress }).getByRole('textbox');
 
     expect(onKeyPress).not.toHaveBeenCalled();
 
-    component.prop('onKeyPress')('q');
+    await user.type(input, 'a');
 
-    expect(onKeyPress).toHaveBeenCalledWith('q');
+    expect(onKeyPress).toHaveBeenCalledWith('a');
   });
 
```

Comment:
Unless I missed it this seems like exactly the same tests that `NxTextInput` has.  The stateful component should have tests for its state handling - that when the user types in the input its `value` and validation change accordingly.

Reply:
Updated here 8fae95e4f6c234f4b80c51c634b3ee8b4ef02e95.